File: Go/ExampleTests/common/request_string.go
package common

type RequestString struct {
	Method string
	Page string
	Address string
	Benchmark string
	Format string
}

func NewRequestStringFromSlice(v []string) RequestString {
	s := RequestString{}
	if len(v) > 0 { s.Method = v[0] }
	if len(v) > 1 { s.Page = v[1] }
	if len(v) > 2 { s.Address = v[2] }
	if len(v) > 3 { s.Benchmark = v[3] }
	if len(v) > 4 { s.Format = v[4] }
	return s
}

// NewRequestStringFromText builds from the text form, e.g. Money as "25 USD".
func NewRequestStringFromText(text string) RequestString {
	parts := RequireTokens(text, 5, "Request")
	return RequestString{
		Method: parts[0],
		Page: parts[1],
		Address: parts[2],
		Benchmark: parts[3],
		Format: parts[4],
	}
}

func (s RequestString) String() string {
	return Token(s.Method) + " " + Token(s.Page) + " " + Token(s.Address) + " " + Token(s.Benchmark) + " " + Token(s.Format)
}

func (s RequestString) Equals(o RequestString) bool {
	return DNCEqual(s.Method, o.Method) &&
		DNCEqual(s.Page, o.Page) &&
		DNCEqual(s.Address, o.Address) &&
		DNCEqual(s.Benchmark, o.Benchmark) &&
		DNCEqual(s.Format, o.Format)
}

func EqualRequestStringSlices(a, b []RequestString) bool {
	if len(a) != len(b) { return false }
	for i := range a {
		if !a[i].Equals(b[i]) { return false }
	}
	return true
}

File: Rust/ExampleTests/common/request_string.rs
#![allow(dead_code, unused_imports, unused_variables)]

#[derive(Debug, Clone, Default)]
pub struct RequestString {
    pub method: String,
    pub page: String,
    pub address: String,
    pub benchmark: String,
    pub format: String,
}

impl RequestString {
    pub fn from_vec(v: &[&str]) -> Self {
        Self {
            method: v.get(0).copied().unwrap_or("").to_string(),
            page: v.get(1).copied().unwrap_or("").to_string(),
            address: v.get(2).copied().unwrap_or("").to_string(),
            benchmark: v.get(3).copied().unwrap_or("").to_string(),
            format: v.get(4).copied().unwrap_or("").to_string(),
        }
    }

    /// Builds from the text form, e.g. Money as "25 USD".
    pub fn from_text(text: &str) -> Self {
        let parts = crate::common::tokens::require(text, 5, "Request");
        Self {
            method: parts[0].clone(),
            page: parts[1].clone(),
            address: parts[2].clone(),
            benchmark: parts[3].clone(),
            format: parts[4].clone(),
        }
    }
}

impl std::fmt::Display for RequestString {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        write!(f, "{}", [crate::common::tokens::token(&self.method), crate::common::tokens::token(&self.page), crate::common::tokens::token(&self.address), crate::common::tokens::token(&self.benchmark), crate::common::tokens::token(&self.format)].join(" "))
    }
}

impl PartialEq for RequestString {
    fn eq(&self, other: &Self) -> bool {
        crate::common::dnc_equal(&self.method, &other.method)
            && crate::common::dnc_equal(&self.page, &other.page)
            && crate::common::dnc_equal(&self.address, &other.address)
            && crate::common::dnc_equal(&self.benchmark, &other.benchmark)
            && crate::common::dnc_equal(&self.format, &other.format)
    }
}
